lib/galaxy/tools/actions/upload_common.py

```diff
@@ -282,8 +282,30 @@ def create_paramfile( trans, uploaded_datasets ):
     """
     def _chown( path ):
         try:
-            pwent = pwd.getpwnam( trans.user.email.split('@')[0] )
-            cmd = [ '/usr/bin/sudo', '-E', trans.app.config.external_chown_script, path, pwent[0], str( pwent[3] ) ]
+            # get username from email/username
```

Comment:
Indeed. Would the helper script solve this problem already? Otherwise could you be more precise what should be packed in a function (just getting the user or also calling the external scripts). Where would such a function fit? I.e. in which python module?